### PR TITLE
Add ngsim documentation

### PR DESCRIPTION
**Pull request by Gamenot**
See CHANGELOG.md

### CODE REVIEW

File: smarts/waymo/README.md [hunk 1]
```diff
@@ -107,13 +107,7 @@ TfRecord Explorer
...
      19  69ce11e9b69203e0           198               51                     198                      2  []            []
      20  31e3acf12ee52d0c           198               77                     198                      2  []            []
-```
-.\
-.\
-.
-
-
-```
+     ...
```
Comment (Adaickalavan) [inline smarts/waymo/README.md:110]: Vertical dots might look better.
```suggestion
      .
      .
      .
```

**Review by qianyi-sun — APPROVED** [review]

File: docs/ecosystem/ngsim.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+        for o, rgb in obs:
+            ...
+
+
+Alternatively, an approach like :examples:`traffic_histories_vehicle_replacement.py` can be used to operate directly with the scenarios.
+
+
+Samples
+------
```
Comment (Adaickalavan) [inline docs/ecosystem/ngsim.rst:98]: Title underline is short.
```suggestion
Samples
-------
```

File: docs/ecosystem/ngsim.rst [hunk 3]
```diff
@@ -0,0 +1,100 @@
+.. _ngsim:
```
Comment (Adaickalavan) [inline docs/ecosystem/ngsim.rst:1]: Document isn't included in any toctree. We may include it under the ecosystem toctree in index.rst .

File: docs/ecosystem/ngsim.rst [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+These scenarios are located in the `smarts repository <https://github.com/huawei-noah/SMARTS/tree/master/scenarios/NGSIM>`_.
+
+To use these scenarios, you must first download the dataset in an appropriate format.
+SMARTS assumes that the data will be in tab-separated text files.
+Its import scripts were tested on and work with the data downloaded from `here <http://bit.ly/PPUU-data>`_,
+which is the same as was used by this `PPUU project <https://github.com/Atcold/pytorch-PPUU>`_.
+
+After the dataset has been downloaded, you will need to update each ``scenario.py``
+in the ngsim scenario directories (e.g., :scenarios:`NGSIM/i80/scenario.py`, :scenarios:`NGSIM/us101/scenario.py`, and/or :scenarios:`NGSIM/peach/scneario.py`) to point the `input_path` field
```
Comment (Adaickalavan) [inline docs/ecosystem/ngsim.rst:22]:
```suggestion
in the ngsim scenario directories (e.g., :scenarios:`NGSIM/i80/scenario.py`, :scenarios:`NGSIM/us101/scenario.py`, and/or :scenarios:`NGSIM/peachtree/scenario.py`) to point the `input_path` field
```

File: docs/ecosystem/ngsim.rst [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+to the dataset location on your filesystem.
+
+Once that is done, you should be able to build your scenarios in the normal way, for example:
+```bash
+scl scenario build-all --clean scenarios/NGSIM
+```
```
Comment (Adaickalavan) [inline docs/ecosystem/ngsim.rst:28]:
```suggestion
Once that is done, you should be able to build your scenarios in the normal way, for example: ``scl scenario build-all --clean scenarios/NGSIM``
```

**Review by Adaickalavan — APPROVED** [review]

File: docs/ecosystem/ngsim.rst [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+For each traffic history dataset specified in your ``scenario.py``, 
+this will create a corresponding ``.shf`` file that SMARTS will use
+whenever this traffic will be added to a simulation.
+
+Note that the SUMO maps created for NGSIM were made by hand by the open-source SMARTS team.
+Although they attempt to align with the positions in the traffic dataset,
+their level of exactness may not be enough for some model-training situations,
+so you may want or need to refine them with SUMO's `netedit tool <https://sumo.dlr.de/docs/Netedit/index.html>`_.
+
+An example of how traffic history might be saved as observations can be found in :py:module:`smarts.dataset.traffic_histories_to_observations`.
```
[the remark this answers is not in the document]
Comment (Gamenot) [reply docs/ecosystem/ngsim.rst:39]: Do you happen to have the documentation that mentions the details of this syntax? I have found that it is possible from some sources but not the list of reference types.

File: docs/ecosystem/ngsim.rst (hunk 3 above)
Comment (Gamenot) [reply docs/ecosystem/ngsim.rst:1]: Done.

File: docs/ecosystem/ngsim.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (Adaickalavan) [reply docs/ecosystem/ngsim.rst]: Cross-referencing Python objects
https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-python-objects

**Pull request merged**